common_helper_files/fail_safe_file_operations.py

```diff
@@ -154,54 +130,49 @@ def get_safe_name(file_name, max_size=200, valid_characters='abcdefghijklmnopqrs
     return safe_name
 
 
-def get_files_in_dir(directory_path):
+def get_files_in_dir(directory_path: Union[str, Path]) -> List[str]:
     '''
     Returns a list with the absolute paths of all files in the directory directory_path
 
     :param directory_path: directory including files
-    :type directory_path: str
-    :return: list
     '''
     result = []
     try:
         for file_path, _, files in os.walk(directory_path):
             for file_ in files:
-                result.append(os.path.abspath(os.path.join(file_path, file_)))
+                result.append(str(Path(file_path, file_).absolute()))
     except Exception as e:
         logging.error('Could not get files: {} {}'.format(sys.exc_info()[0].__name__, e))
     return result
 
 
-def get_dirs_in_dir(directory_path):
+def get_dirs_in_dir(directory_path: Union[str, Path]) -> List[str]:
     '''
     Returns a list with the absolute paths of all 1st level sub-directories in the directory directory_path.
 
     :param directory_path: directory including sub-directories
-    :type directory_path: str
-    :return: list
     '''
     result = []
     try:
-        dir_content = os.listdir(directory_path)
-        for item in dir_content:
-            dir_path = os.path.join(directory_path, item)
-            if os.path.isdir(dir_path):
-                result.append(dir_path)
+        p = Path(directory_path)
+        for item in p.iterdir():
+            if not Path(item).is_dir():
+                continue
+            result.append(str(item.resolve()))
```

Comment:
The `continue` is not needed if you invert the logic. Also the `Path(item)` is not necessary, because `iterdir()` yields `Path` objects.
```suggestion
        for item in p.iterdir():
            if item.is_dir():
                result.append(str(item.resolve()))
```

Can also be written as list comprehension:
```suggestion
        return [str(item.resolve()) for item in p.iterdir() if item.is_dir()]
```

Reply:
Not a huge fan of list comprehensions so I inverted the logic but didn't use a comprehension.